ros/api/v1/recommendation_ratings.py

```diff
@@ -28,20 +28,28 @@ def post(self):
 
         data = json.loads(request.data)
         inventory_id = data['inventory_id']
-        rating = int(data['rating'])
-        if rating not in [c.value for c in RatingChoicesEnum]:
-            abort(
-                422,
-                message=(
-                    "{} is not a valid value for rating".format(rating)))
+        allowed_choices = [c.value for c in RatingChoicesEnum]
+
+        def invalid_value_msg(val):
+            return (
+                f"'{val}' is invalid value for rating."
+                f'Possible values - { *allowed_choices, }'
+            )
+        rating = None
+        try:
+            rating = int(data['rating'])
+        except ValueError:
+            abort(400, message=(invalid_value_msg(data['rating'])))
+
+        if rating not in allowed_choices:
+            abort(422, message=(invalid_value_msg(data['rating'])))
```

Comment:
Please let me know your thoughts on this `422` error code. Do we want to keep it as it is or you want me to change.

Reply:
As per the docs : https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/422 the 422 is good fit here!